File: java/com/leaplight/LeapLight.java
package com.leaplight;

import com.leaplight.hue.HueAdapter;
import com.leaplight.listener.MainListener;
import com.leapmotion.leap.Controller;
import java.io.IOException;

public final class LeapLight {

  public static void main(String[] args) {
    System.out.println("Press Enter to quit...");

    HueAdapter adapter = new HueAdapter();

    MainListener listener = new MainListener();

    Controller controller = new Controller();
    controller.addListener(listener);

    try {
      System.in.read();
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}

File: leaplight.py
import sys
from lib import Leap

class Listener(Leap.Listener):
    def on_connect(self, controller):
        print "Connected"

    def on_frame(self, controller):
        # print "Frame available"
        frame = controller.frame()

        print "Frame id: %d, timestamp: %d, hands: %d, fingers: %d" % (
          frame.id, frame.timestamp, len(frame.hands), len(frame.fingers))
        pass

def main():

    listener = Listener()
    controller = Leap.Controller()
    controller.add_listener(listener)

    # Keep this process running until Enter is pressed
    print "Press Enter to quit..."
    try:
        sys.stdin.readline()
    except KeyboardInterrupt:
        pass

if __name__ == "__main__":
    main()
